src/EnergyPlus/WeatherManager.cc

```diff
@@ -6946,54 +6075,33 @@ namespace WeatherManager {
+                                                                               {DDHumIndType::Enthalpy, "Enthalpy [J/kg]"},
+                                                                               {DDHumIndType::HumRatio, "Humidity Ratio []"},
+                                                                               {DDHumIndType::RelHumSch, "Schedule []"},
+                                                                               {DDHumIndType::WBProfDef, "WetBulbProfileDefaultMultipliers []"},
+                                                                               {DDHumIndType::WBProfDif, "WetBulbProfileDifferenceSchedule []"},
+                                                                               {DDHumIndType::WBProfMul, "WetBulbProfileMultiplierSchedule []"}};
+
         // Below are the 2009 fractions, HOF, Chap 14, Table 6
         static Array1D<Real64> const DefaultTempRangeMult(24, {0.88, 0.92, 0.95, 0.98, 1.0,  0.98, 0.91, 0.74, 0.55, 0.38, 0.23, 0.13,
                                                                0.05, 0.00, 0.00, 0.06, 0.14, 0.24, 0.39, 0.50, 0.59, 0.68, 0.75, 0.82});
 
         // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
-        int EnvrnNum;    // Environment Loop to pass to Design Day Setup Routine
-        int NumAlpha;    // Number of material alpha names being passed
-        int NumNumerics; // Number of material properties being passed
-        int IOStat;      // IO Status when calling get input subroutine
-        int HrLoop;
-        int TSLoop;
-        Real64 LastHrValue;
-        Real64 WNow;
-        Real64 WPrev;
-        Real64 testval;
-        bool errFlag;
-        int DDLoop;
-        std::string envTitle;
         std::string units;
         OutputProcessor::Unit unitType;
-        int schPtr;
-        bool MaxDryBulbEntered;
-        bool PressureEntered;
-
-        // FLOW:
 
         DesDayInput.allocate(TotDesDays); // Allocate the array to the # of DD's
-        DDDBRngModifier.allocate(NumOfTimeStepInHour, 24, TotDesDays);
+        DDDBRngModifier.allocate(DataGlobals::NumOfTimeStepInHour, 24, TotDesDays);
         DDDBRngModifier = 0.0;
-        DDHumIndModifier.allocate(NumOfTimeStepInHour, 24, TotDesDays);
+        DDHumIndModifier.allocate(DataGlobals::NumOfTimeStepInHour, 24, TotDesDays);
         DDHumIndModifier = 0.0;
-        DDBeamSolarValues.allocate(NumOfTimeStepInHour, 24, TotDesDays);
+        DDBeamSolarValues.allocate(DataGlobals::NumOfTimeStepInHour, 24, TotDesDays);
         DDBeamSolarValues = 0.0;
-        DDDiffuseSolarValues.allocate(NumOfTimeStepInHour, 24, TotDesDays);
+        DDDiffuseSolarValues.allocate(DataGlobals::NumOfTimeStepInHour, 24, TotDesDays);
         DDDiffuseSolarValues = 0.0;
-        DDSkyTempScheduleValues.allocate(NumOfTimeStepInHour, 24, TotDesDays);
+        DDSkyTempScheduleValues.allocate(DataGlobals::NumOfTimeStepInHour, 24, TotDesDays);
```

Comment:
It definitely feels like these should be a singly allocated struct rather than individual arrays.  There are certainly parts of the code where we iterate on objects of an array in a tight loop, and in those cases, we benefit from having individual arrays of values.  But that is definitely not the case here.  It would be easier to maintain if these were just member variables of a struct which was allocated one time.  I still don't think it's necessary to add more changes to do that here since it appears this is done, but just forward thinking.

Reply:
Good point.